src/compiler/checker.ts

```diff
@@ -4703,7 +4799,10 @@ namespace ts {
                 return result;
             }
 
-            function propertiesIdenticalTo(source: ObjectType, target: ObjectType): Ternary {
+            function propertiesIdenticalTo(source: Type, target: Type): Ternary {
+                if (!(source.flags & TypeFlags.ObjectType && target.flags & TypeFlags.ObjectType)) {

```

Comment:
We already break down unions and intersections elsewhere. Here we only want to compare identical kinds of types.